Tests/Integration/RedisFixture.cs

```diff
@@ -124,17 +125,32 @@ public async Task InitializeAsync()
             await StartRedisContainer();
 
             var redisConnectionString = $"localhost:{RedisPort}";
-            try
+            var isRedisReady = false;
+            var maxNumberOfRetries = 10;
+            var tryRun = 0;
+            while (!isRedisReady && tryRun < maxNumberOfRetries)
             {
-                this.Redis = await ConnectionMultiplexer.ConnectAsync(redisConnectionString);
-                Database = this.Redis.GetDatabase();
+                try
+                {
+                    Redis = await ConnectionMultiplexer.ConnectAsync(redisConnectionString);
+                    Database = this.Redis.GetDatabase();
+                    isRedisReady = true;
+                }
+                catch (Exception ex)
+                {
+                    await Task.Delay(1000);
+                    Console.WriteLine($"Failed to connect to redis at '{redisConnectionString}'. If running locally with docker: run 'docker run -d -p 6379:6379 redis'. If running in Azure DevOps: run redis in docker.", ex);
+                }
+
+                tryRun++;
```

Comment:
nit: we can put the tryRun++ inside the catch block, to not have it executed when try is succeeding

Reply:
Good catch thanks